test/forms/inputs/color-input.test.js

```diff
@@ -1,32 +1,47 @@
-import React from 'react'
-import { mount } from 'enzyme'
+import React, { useState } from 'react'
 import { ColorInput } from '../../../src/'
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { within } from '@testing-library/dom'
 
-test('ColorInput hex input adds hash to value', () => {
-  const onChange = jest.fn()
-  const props = {
+const WrappedColorInput = (props) => {
+  const [value, setValue] = useState('')
+
+  const defaultProps = {
     input: {
       name: 'test',
-      value: '',
-      onChange,
+      value: value,
+      onChange: (e) => setValue(e),
     },
     meta: {},
   }
-  const wrapper = mount(<ColorInput {...props} />)
-  wrapper.find('.hex-input').simulate('change', { target: { value: '000' } })
-  expect(onChange).toHaveBeenCalledWith('#000')
+
+  return <ColorInput {...defaultProps} {...props} />
+}
+
+test('ColorInput hex input adds hash to value', async () => {
+
+  render(<WrappedColorInput />)
+
+  const input = screen.getByRole('textbox')
+
+  const user = userEvent.setup()
+
+  await user.click(input)
+
+  await user.keyboard('{0}{0}{0}{Enter}');
+
+  expect(input).toHaveValue('000')
 })
 
-test('ColorInput expands dropdown when hex input is focused', () => {
-  const props = {
-    input: {
-      name: 'test',
-      value: '',
-    },
-    meta: {},
-  }
-  const wrapper = mount(<ColorInput {...props} />)
-  expect(wrapper.find('.popover').exists()).toBe(false)
-  wrapper.find('.hex-input').simulate('focus')
-  expect(wrapper.find('.popover').exists()).toBe(true)
+test('ColorInput expands dropdown when hex input is focused', async () => {
+  const { container } = render(<WrappedColorInput />)
+
+  const input = screen.getByRole('textbox')
+
+  const user = userEvent.setup()
+
+  await user.click(input)
+
+  expect(within(container).getByText('hex')).toBeTruthy()
```

Comment:
In this specific test, `getByText` would throw if not found, so there's no difference in practice.

Since we're testing HTML elements though, I vote for `toBeInTheDocument()` to be the default. The element must be added to the DOM to pass this validation, while the other matchers just test for the existence of the element.

Reply:
@chawes13 Also, I opted for adding a "dialog" role to the popover element in the `ColorPicker` component. Does that sound right?